src/library/scala/util/ChainingOps.scala

```diff
@@ -0,0 +1,45 @@
+package scala
+package util
+
+trait ChainingSyntax {
+  implicit final def scalaUtilChainingOps[A](a: A): ChainingOps[A] = new ChainingOps(a)
+}
+
+/** Adds chaining methods `tap` and `pipe` to every type.
+ */
+final class ChainingOps[A](val self: A) extends AnyVal {
+  /** Applies `f` to the value for its side effects, and returns the original value.
+   *
+   *    {{{
+   *    val xs = List(1, 2, 3)
+   *               .tap(ys => println("debug " + ys.toString))
+   *    // xs == List(1, 2, 3)
+   *    }}}
+   *
+   *  @param f      the function to apply to the value.
+   *  @tparam U     the result type of the function `f`.
+   *  @return       the original value `self`.
+   */
+  def tap[U](f: A => U): self.type = {
+    f(self)
+    self
+  }
+
+  /** Converts the value by applying the function `f`.
+   *
+   *    {{{
+   *    val times6 = (_: Int) * 6
+   *    val i = (1 - 2 - 3).pipe(times6).pipe(scala.math.abs)
+   *    // i == 24
+   *    }}}
+   *
+   * Note: `(1 - 2 - 3).pipe(times6)` may have a small amount of overhead at
+   * runtime compared to the equivalent  `{ val temp = 1 - 2 - 3; times6(temp) }`.
+   *
+   *  @param f      the function to apply to the value.
+   *  @tparam B     the result type of the function `f`.
+   *  @return       a new value resulting from applying the given function
+   *                `f` to this value.
+   */
+  def pipe[B](f: A => B): B = f(self)
```

Comment:
Other languages name this `|>`. Would be great to do so here too.

Reply:
Mentioned here: https://github.com/scala/scala/pull/6767#issuecomment-410906305

Reply:
Ideally if we can ignore backward compat is get rid of `match` as keyword and make it a function named `match`. `|>` is basically `match`.

Reply:
Seconding @japgolly . I'm not for symbols in general but this is a very very special ubiquitous one. bash uses something similar, F# has it, PowerShell too.

Reply:
@ScalaWilliam You can add OCaml & ReasonML to that list.

Reply:
Maybe one of you should send a new pull request or contributor thread and get consensus around it.

Reply:
As advised by @eed3si9n , here's a PR: https://github.com/scala/scala/pull/7326

fyi @japgolly @RawToast @hepin1989